From: KWSys Robot <kwrobot@kitware.com>
Date: Tue, 15 Oct 2013 12:43:32 +0000 (-0400)
Subject: KWSys 2013-10-15 (6eab64c3)
X-Git-Tag: v3.0.0-rc1~178^2^2~5
X-Git-Url: http://cmake.org/gitweb?p=cmake.git;a=commitdiff_plain;h=c01e7445

KWSys 2013-10-15 (6eab64c3)

Extract upstream KWSys using the following shell commands.

$ git archive --prefix=upstream-kwsys/ 6eab64c3 | tar x
$ git shortlog --no-merges --abbrev=8 --format='%h %s' 96b2f6f4..6eab64c3
Raphael Kubo da Costa (1):
      6eab64c3 SystemInformation: Include backtrace-related headers on FreeBSD

Change-Id: Ib0447aab3d6555ce278f0c1417af2c53e6f181a5
---

diff --git a/Source/kwsys/SystemInformation.cxx b/Source/kwsys/SystemInformation.cxx
index 7c31f3a..2672730 100644
--- a/Source/kwsys/SystemInformation.cxx
+++ b/Source/kwsys/SystemInformation.cxx
@@ -88,6 +88,15 @@ typedef int siginfo_t;
 #  include <ifaddrs.h>
 #  define KWSYS_SYSTEMINFORMATION_IMPLEMENT_FQDN
 # endif
+# if defined(KWSYS_SYSTEMINFORMATION_HAS_BACKTRACE)
+#  include <execinfo.h>
+#  if defined(KWSYS_SYSTEMINFORMATION_HAS_CPP_DEMANGLE)
+#    include <cxxabi.h>
+#  endif
+#  if defined(KWSYS_SYSTEMINFORMATION_HAS_SYMBOL_LOOKUP)
+#    include <dlfcn.h>
+#  endif
+# endif
 #endif
 
 #if defined(__OpenBSD__) || defined(__NetBSD__)
